### DISCUSSION TITLE
Initialize Project With Error in Next.js source code

### DISCUSSION THREAD

**Discussion by aethony** [vercel/next.js#50954] [Help] [answered]
### Summary

I am creating new project with npx create-next-app@latest. After the project is successfully created, when I use npm to start the project or use the next dev command to start the project, an error will be reported that the source code inheritance class inherits an undefined value. How should I handle this situation, seek help. There are two methods I tried:
(1) delete the entire node_modules folder, then npm install the total packages
(2) lower the version of webpack package
Neither of method works
![image](https://github.com/vercel/next.js/assets/46308957/5c0fa115-1fa5-49ad-8ba6-a835e7f4c71e)
The package info is here:
![image](https://github.com/vercel/next.js/assets/46308957/7da2cc04-5fdb-4160-82d4-aa0bf7e75d66)

**Comment by icyJoseph** [comment]
What's your Node.js version? Try to make sure it is at least Node 16.8.x

## Edit

The minimum Node.js version is 16.8:

- https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#nodejs-version

> The minimum Node.js version is now v16.8. See the [Node.js documentation](https://nodejs.org/docs/latest-v16.x/api/) for more information.

**Reply by aethony** [reply]
Yep, my node version is 16.7.0. But it does not work.

**Reply by icyJoseph** [reply]
On another note, even though Next.js uses webpack, is it not the same way we do with other dependencies. Once you have fixed your node version, remove the webpack 4.0.0, you installed, and then you can run `npm explain webpack` or `yarn why webpack`. You'd think it would show webpack 5, but it'll say something like this:

```console
npm ERR! No dependencies found matching webpack
```

That's because Next.js has a compiled version of webpack, built in! 

- https://github.com/vercel/next.js/tree/canary/packages/next/src/compiled

In fact, a whole bunch of dependencies are pre-compiled. 

Anyhow, I believe the issue must have been your Node.js version. If not, we can look further after that.

## Breaking

One thing though, webpack 4 to 5 is not just a matter of downgrading, there's a few non-compatible changes. So please, remove that webpack 4 dependency. As I've shown you, Next.js already has a built-in webpack 5.

**Reply by icyJoseph** [reply]
@aethony which settings did you use? JavaScript w/ App Router? I am making a fresh project right now to see if I can reproduce the issue.

**Reply by icyJoseph** [reply]
@aethony looks like 16.7.0 has some issue, I've tried with 16.16.0 and it worked fine.

**Reply by aethony** [reply]
use js.I have upgrading the webpack to the version of latest. Also, it does not work.
![image](https://github.com/vercel/next.js/assets/46308957/d3d0a534-1f64-4834-91de-a473b9275430)

**Reply by icyJoseph** [reply]
It is a problem with that Node.js version, I've tried with both 16.16.0 - works, but using 16.7.0 crashes. Stop adding webpack, remove it!

**Reply by icyJoseph** [reply]
It works with 16.8.0 as well, and fails with 16.6.0, so it must have been something not available up to 16.8.x -

**Reply by aethony** [reply]
So i need upgrading the version of node to 16.16.0?

**Reply by icyJoseph** [reply]
Yeah, anything past the one you have, tough luck! I'd recommend taking Node 18.x.x altogether, since the LTS is Node 18.16.x https://nodejs.org/en

**Reply by icyJoseph** [reply, marked as answer]
The docs have this bit as well: https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#nodejs-version

> The minimum Node.js version is now v16.8. See the [Node.js documentation](https://nodejs.org/docs/latest-v16.x/api/) for more information.

**Reply by aethony** [reply]
Yep, it work now. But it not declared which version of Node that can solve this problem.In this way, only each version of node could be tried. Node Version up to 18.16 is ok.
Thank u so much
